src/requestVariableCacheValueProcessor.ts

```diff
@@ -0,0 +1,109 @@
+"use strict";
+import { window } from "vscode";
+
+import { MimeUtility } from "./mimeUtility";
+import { RequestVariableCacheValue } from './models/requestVariableCacheValue';
+import { HttpResponse } from './models/httpResponse';
+
+const validPathRegex = new RegExp(/(\w+)(\.\w+|\[\d+\])*/);
+const partRegex = new RegExp(/(\w+)(\[\d+\])*/);
+const arrayIndexRegex = new RegExp(/\[(\d+)\]/);
+
+export class RequestVariableCacheValueProcessor {
+    public static getValueAtPath(value: RequestVariableCacheValue, path: string): any {
+        if (!value
+            || !path
+            || !RequestVariableCacheValueProcessor.isValidPath(path)) {
+            return undefined;
+        }
+
+        const parts = path.split(".");
+
+        // Expect first part to be variable name
+        if (parts.length === 1) {
+            return value;
+        }
+
+        switch (parts[1]) {
+            case "request":
+                return RequestVariableCacheValueProcessor.resolveParts(
+                    value.request,
+                    parts.slice(2, parts.length)
+                );
+            case "response":
+                const { response } = value;
+                return RequestVariableCacheValueProcessor.resolveResponse(response, parts);
+            default:
+                return undefined;
+        }
+    }
+
+    private static resolveResponse(response: HttpResponse, parts: string[]) {
+        if (parts.length >= 3) {
+            const matches = parts[2].match(partRegex);
+            // Must parse body specifically
+            if (matches[1] === "body") {
+                let bodyValue = RequestVariableCacheValueProcessor.parseResponseBody(response);
+                // If array indeces:
+                for (let j = 2; j < matches.length; j++) {
```

Comment:
Do we support json path here, if so, can we use some existing packages on npm to do such kind of parsing/querying job?

Reply:
Well, I looked into this, and couldn't quite decide how to solve it. If there are any suggestions here they are welcome.

The issue, as I saw it, is that not the whole object-structure is a JSON-string. Sure we could allow JSON path for the `body` of `response`, but what about the rest of the object? Should JSON path start from the `body`? 

`variableName.response.body$.users[:1].id`?

I couldn't decide on a good approach, so I just did a simple naive manual resolving of values.

Reply:
@cbrevik my suggestion is that make the syntax like `variableName.response.body.$.users...`
That means the whole syntax is likes:
```
variableName.(response|request).(header|body).(header name|JSON Path)
```